### PR TITLE
Enemy proximity refactor

### LINKED ISSUE #29: More efficient implementation of DistToClosestEnemy
For better performance when there are multiple enemies.

Maybe use Physics.OverlapSphere?

### PR DESCRIPTION
**Pull request by fnnbrr**
More efficient implementation of DistToClosestEnemy in AwakenessManager.cs

Addresses Issue #29

EDIT:  This was done by creating a new "Enemy" component that can be attached to any hostile GameObject; my definition of "hostile" is that the object should increase Birdie's awakeness and restart the game at the last checkpoint upon contact.  All spawned-in enemies are tracked in a public "pool" that can be queried to do things like finding the closest enemy to a position.

### CODE REVIEW

File: Assets/Scripts/AwakenessManager.cs [hunk 1]
```diff
@@ -78,18 +78,12 @@ private void FixedUpdate()
...
-            if (curDistance < minDistance)
-            {
-                minDistance = curDistance;
-            }
-        }
-        return minDistance;
+        if (Enemy.Enemies.Count == 0) return Mathf.Infinity;
+
+        Vector3 playerPosition = transform.position;
+        Vector3 closestEnemyPosition = Enemy.ClosestEnemy(playerPosition).transform.position;
```
Comment (jurcaua) [inline Assets/Scripts/AwakenessManager.cs]: The "ClosestEnemy" function seems to only be used here so... maybe we can just make this "ClosestEnemyDistance" or something, so we don't need the extra distance calculation below?

**Review by jurcaua — CHANGES_REQUESTED** [review]
Just one comment, and rest looks great! Thanks for looking into this!!!! 🚢

File: Assets/Scripts/AwakenessManager.cs (hunk 1 above)
Comment (fnnbrr) [reply Assets/Scripts/AwakenessManager.cs]: Gotcha, updated

File: Assets/Scripts/NPCs/Components/Enemy.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+            
+            foreach (Enemy curEnemy in Enemies)
+            {
+                float curDistance = (curEnemy.transform.position - queryPosition).sqrMagnitude;
+                if (curDistance < minDistance)
+                {
+                    minDistance = curDistance;
+                }
+            }
+            return Mathf.Sqrt(minDistance);
```
Comment (jurcaua) [inline Assets/Scripts/NPCs/Components/Enemy.cs:44]: nice!!!!

**Review by jurcaua — APPROVED** [review]
beauty, lets get it in boysssssss

**Pull request merged**